### ISSUE TITLE
Recording gameplay "demofiles path not specified"

### ISSUE THREAD

**Issue by ryanprinster** [deepmind/lab#112]
Whenever I try to use any of the flags --video, --record, --demo in random_agent.py with any arguments, I get something like

`RuntimeError: Failed to start environment - "Recording failed: demofiles path not specified."
`

How do I specify the demofiles path? And it doesn't seem to be super clear with the documentation, is this supposed to be the argument to these flags (if not what is)?

**Comment by tkoeppe** [comment]
You're right, the random agent seems to fail to pass a mandatory setting to the environment. The following patch exposes the new `--demofiles` flag:

```
diff --git a/python/random_agent.py b/python/random_agent.py
index 39927a8..a1cfab2 100644
--- a/python/random_agent.py
+++ b/python/random_agent.py
@@ -137,7 +137,7 @@ class SpringAgent(object):
     self.action = np.zeros([len(self.action_spec)])
 
 
-def run(length, width, height, fps, level, record, demo, video):
+def run(length, width, height, fps, level, record, demo, demofiles, video):
   """Spins up an environment and runs the random agent."""
   config = {
       'fps': str(fps),
@@ -148,6 +148,8 @@ def run(length, width, height, fps, level, record, demo, video):
     config['record'] = record
   if demo:
     config['demo'] = demo
+  if demofiles:
+    config['demofiles'] = demofiles
   if video:
     config['video'] = video
   env = deepmind_lab.Lab(level, ['RGB_INTERLEAVED'], config=config)
@@ -192,6 +194,8 @@ if __name__ == '__main__':
                       help='Record the run to a demo file')
   parser.add_argument('--demo', type=str, default=None,
                       help='Play back a recorded demo file')
+  parser.add_argument('--demofiles', type=str, default=None,
+                      help='Directory for demo files')
   parser.add_argument('--video', type=str, default=None,
                       help='Record the demo run as a video')
 
@@ -199,4 +203,4 @@ if __name__ == '__main__':
   if args.runfiles_path:
     deepmind_lab.set_runfiles_path(args.runfiles_path)
   run(args.length, args.width, args.height, args.fps, args.level_script,
-      args.record, args.demo, args.video)
+      args.record, args.demo, args.demofiles, args.video)

```

(I think the setting didn't originally exist and demos were written into the "home directory" (in the Quake sense), and when we made an explicit, separate setting for the demo directory, we didn't update the random agent.)

I'll try and commit a similar patch soon.

Thanks for reporting!

**Comment by tkoeppe** [comment]
Fixed by 4293f0338ede63691abac778de1b2801bd9092ff.
